File: 563.java

// Definition for a binary tree node.
public class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;
    TreeNode() {}
    TreeNode(int val) { this.val = val; }
    TreeNode(int val, TreeNode left, TreeNode right) {
        this.val = val;
        this.left = left;
        this.right = right;
    }
}

class Solution {
    int tilt = 0;
    public int findTilt(TreeNode root) {
        dfs(root);
        return tilt;
    }
    private int dfs(TreeNode node) {
        if (node == null) {
            return 0;
        }
        int left = dfs(node.left);
        int right = dfs(node.right);
        if (left > right) {
            this.tilt += left - right;
        }else {
            this.tilt += right - left;
        }
        return node.val + left + right;
    }
}

File: 563.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def findTilt(self, root: TreeNode) -> int:
        total_tilt = 0
        def dfs(node):
            if not node:
                return 0
            left_sum = dfs(node.left)
            right_sum = dfs(node.right)
            nonlocal total_tilt
            total_tilt += abs(left_sum - right_sum)
            return node.val + left_sum + right_sum
        
        dfs(root)
        return total_tilt